### DISCUSSION TITLE
Running medusa admin on github codespaces and gitpod

### DISCUSSION THREAD

**Discussion by blitze** [medusajs/medusa#4822] [Q&A]
I installed medusa using the `yarn create medusa-app --seed`, which installed the backend as well as the admin UI. 

On Gitpod, I am able to pull in products using medusa js client but when I try to access the admin at localhost:7001 I get "Invalid host header"

On Codespaces, I am using a nodejs + postgres container and I am able to pull in products on the storefront using medusa js client but when I try to access the admin at localhost:7001, the page does not load most of the time and it shows  this message in terminal "<i> [webpack-dev-server] Gracefully shutting down. To force exit, press ^C again. Please wait..." Every now and then, the page does load but I can't login because of CORS issues, which I have not been able to resolve by setting the ADMIN_CORS / STORE_CORS. I have also tried using ngrok but ended up with the "Invalid host header" as above.

Anyone had any success running medusa admin on one of these services? Help please.

Thanks

**Comment by yasirmturk** [comment]
Something similar, what is the command to run admin only? I know `medusa develop` starts backend + admin. I can run backend only on 9000 using `node index.js` for example but how can I run admin?? 

Secondly, is it possible to run admin on the same port but different path? for example localhost:9000/a

**Comment by yasirmturk** [comment]
btw I am able to run backend and frontend on docker compose but admin wont run; this is what i can from the logs
```

medusa-server        | <e> [webpack-dev-middleware] HookWebpackError: Bindings not found.
medusa-server        | <e>     at makeWebpackError (/app/medusa/node_modules/webpack/lib/HookWebpackError.js:48:9)
medusa-server        | <e>     at /app/medusa/node_modules/webpack/lib/Compilation.js:3068:12
medusa-server        | <e>     at eval (eval at create (/app/medusa/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:30:1)
medusa-server        | <e>     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
medusa-server        | <e> -- inner error --
medusa-server        | <e> Error: Bindings not found.
medusa-server        | <e>     at Compiler.<anonymous> (/app/medusa/node_modules/@swc/core/index.js:102:19)
medusa-server        | <e>     at Generator.next (<anonymous>)
medusa-server        | <e>     at /app/medusa/node_modules/@swc/core/index.js:34:71
medusa-server        | <e>     at new Promise (<anonymous>)
medusa-server        | <e>     at __awaiter (/app/medusa/node_modules/@swc/core/index.js:30:12)
medusa-server        | <e>     at Compiler.minify (/app/medusa/node_modules/@swc/core/index.js:95:16)
medusa-server        | <e>     at /app/medusa/node_modules/@swc/core/index.js:365:25
medusa-server        | <e>     at Generator.next (<anonymous>)
medusa-server        | <e>     at /app/medusa/node_modules/@swc/core/index.js:34:71
medusa-server        | <e>     at new Promise (<anonymous>)
medusa-server        | <e> caused by plugins in Compilation.hooks.processAssets
medusa-server        | <e> Error: Bindings not found.
medusa-server        | <e>     at Compiler.<anonymous> (/app/medusa/node_modules/@swc/core/index.js:102:19)
medusa-server        | <e>     at Generator.next (<anonymous>)
medusa-server        | <e>     at /app/medusa/node_modules/@swc/core/index.js:34:71
medusa-server        | <e>     at new Promise (<anonymous>)
medusa-server        | <e>     at __awaiter (/app/medusa/node_modules/@swc/core/index.js:30:12)
medusa-server        | <e>     at Compiler.minify (/app/medusa/node_modules/@swc/core/index.js:95:16)
medusa-server        | <e>     at /app/medusa/node_modules/@swc/core/index.js:365:25
medusa-server        | <e>     at Generator.next (<anonymous>)
medusa-server        | <e>     at /app/medusa/node_modules/@swc/core/index.js:34:71
medusa-server        | <e>     at new Promise (<anonymous>)
```

**Reply by blackavec** [reply]
i'm getting the same thing, how did you fix it? i don't get it on mac, only when i run it in docker

**Reply by OliCDev** [reply]
I'm getting the exact same error; has anyone found a solution to this?

**Reply by Inerska** [reply]
Same here, any solution ?

**Reply by robin-grenzdoerfer** [reply]
same issue -.-

**Comment by pmltechpile** [comment]
I am also experiencing the same issue. I've setup the medusa backend + admin on an ec2 instance on aws. I'm using cloudflare as a reverse proxy into the ec2 and I'm able to reach the backend with calls from my separate frontend but when I try and access the admin I get the same `Invalid Host header` error. I've tried just about every suggestion I've ran across on SO but nothing seems to work. Hoping someone can comment with an idea that'll lead me in the right direction because I'm completely in the dark with this one.

**Reply by DukeRupert** [reply]
You can find this in the MedusaJS docs [here](https://docs.medusajs.com/admin/configuration)

Under admin config options add your admin url to allowed hosts or set it to "all". This fixes the Invalid `Host Header error` for me but I still run into a `net::ERR_CONNECTION_REFUSED` after that. 

/medusa-config.js
```
{
 resolve: "@medusajs/admin",
  /** @type {import('@medusajs/admin').PluginOptions} */
  options: {
    autoRebuild: true,
    develop: {
      open: process.env.OPEN_BROWSER !== "false",
      allowedHosts: "all",
    },
  },
},
```

**Reply by maximbroos** [reply]
Do you remember what you did to fix the ERR_CONNECTION_REFUSED issues? Having a similar issue...

**Comment by HumansIt** [comment]
If you run medusa in mode production, you can access to the admin ui by the url http://localhost:9000/app

**http://localhost:9000** is the url and port of your backend project

**/app** is the default value set in medusa-config.js options by the path variable


[https://docs.medusajs.com/admin/quickstart#production-path](url)
